File: Assignment_46X/Assignment_46.3.cpp
#include<iostream>
using namespace std;

struct node
{
    int data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node *PNODE;

class SinglyLL
{
    private:
        PNODE first;
        int iCount;

    public:

        SinglyLL()
        {
            cout<<"Object of SinglyLL gets created"<<"\n";
            first = NULL;
            iCount = 0;
        }

    void InsertFirst(int no)
    {
        PNODE newn = NULL;

        newn = new NODE;

        newn->data = no;
        newn->next = NULL;

        if(first == NULL)
        {
            first = newn;
        }
        else
        {
            newn->next = first;
            first = newn;
        }
    }

    void Display()
    {
        PNODE temp = NULL;

        temp = first;

        while(temp != NULL)
        {
            cout<<"| "<<temp->data<<" |->";
            temp = temp->next;
        }

        cout<<"NULL\n";
    }

    bool IsEmpty()
    {
        PNODE temp = NULL;
         
        temp = first;

           if((temp->data) == NULL)
            {
                return true;
            }
            else
            {
                return false;
            }
            
            temp = temp->next;
               
        }
    
};

int main()
{
    SinglyLL obj;
    bool bRet = false;

    obj.InsertFirst(51);
    obj.InsertFirst(40);
    obj.InsertFirst(31);
    obj.InsertFirst(20);
    obj.InsertFirst(11);
    obj.InsertFirst(31);

    obj.Display();

    bRet = obj. IsEmpty();
    
    if(bRet == true)
    {
        cout<<"Linked List is Empty"<<"\n";
    } 
    else
    {
        cout<<"Linked List is not Empty"<<"\n";
    }    
    return 0;
}

File: Assignment_46XX/Assignment_46_3.java
class node
{
    public int data;
    public node next;
    
    public node(int no)
    {
        data = no;
        next = null;
    }
}

class SinglyLL
{
    private node first;
    private int iCount;

    public SinglyLL() 
    {
        System.out.println("Object Of SinglyLL gets created");
        first = null;
        iCount = 0;
    }

    public void InsertFirst(int no)
    {
        node newn  = null;

        newn = new node(no);

        if(first == null)
        {
            first = newn;
        }
        else
        {
            newn.next = first;
            first = newn;
        }
    }
    public void Display()
    {
        node temp = null;

        temp = first;

        while(temp != null)
        {
            System.out.print("| "+temp.data+" |->");
            temp = temp.next;
        }

        System.out.println("null\n");
    }

    public boolean  IsEmpty()
    {
        node temp = null;

        temp = first;

            if(temp == null )
            {
                
                return false;
            }
            else
            {
                return true;
            }
        
    }

    
}

class Assignment_46_3
{
    public static void main(String Arr[])
    {
        SinglyLL obj;

        boolean bRet = false;

        obj = new SinglyLL();

        obj.InsertFirst(51);
        obj.InsertFirst(41);
        obj.InsertFirst(31);
        obj.InsertFirst(10);
        obj.InsertFirst(10);
        obj.InsertFirst(24);
        obj.InsertFirst(10);
        obj.InsertFirst(4);

        obj.Display();

        bRet = obj.IsEmpty();

        if(bRet == true)
        {
            System.out.println("Linked List is Not Empty");
        }
        else
        {
            System.out.println("Linked List Empty");
        }

        obj = null;

        System.gc();
    }
}